### ISSUE TITLE
Computation of the masked accuracy

### ISSUE THREAD

**Issue by hankes-garden** [tkipf/gcn#34]
Hi, I notice that, in metrics.py, the accuracy is computed as:
****
def masked_accuracy(preds, labels, mask):
    """Accuracy with masking."""
    correct_prediction = tf.equal(tf.argmax(preds, 1), tf.argmax(labels, 1))
    accuracy_all = tf.cast(correct_prediction, tf.float32)
    mask = tf.cast(mask, dtype=tf.float32)
    mask /= tf.reduce_mean(mask)
    accuracy_all *= mask
    return tf.reduce_mean(accuracy_all)
****
Here, the accuracy_all is first multiplied with the mean of masks over all samples, and then its values are averaged again (again over all samples). I don't quite understand why the average is done over all samples? In particular, I thought we should only average over the the training samples?

As I'm solving only a classification problem, not semi-supervised learning, I thought the accuracy should be:
****
def masked_accuracy(preds, labels, mask):
    """Accuracy with masking for supervised classfisication."""
    correct_prediction = tf.equal(tf.argmax(preds, 1), tf.argmax(labels, 1))
    accuracy_all = tf.cast(correct_prediction, tf.float32)
    mask = tf.cast(mask, dtype=tf.float32)
    accuracy_all *= mask
    returen tf.reduce_sum(accuracy_all)/tf.reduce_sum(mask)
****
Is this correct?

**Comment by tkipf** [comment]
Note that the first operation is a division by the mean. This is meant to
rescale the average accuracy as it would otherwise be influenced by the
unlabeled examples.

**Comment by hankes-garden** [comment]
Oh my bad. I make a mistake on the first operation and thought it is divide by sum. Thanks for your quick reply.
